### PR TITLE
Add option to display mountpoint names instead of filesystem names.

### PR DESCRIPTION
**Pull request by jaunix**
Add option to show mountpoint names instead of filesystem device, via "sar -F MOUNT".  Useful when some crazy fool has used generic LV names.

Existing output as below:

    $ ./sadc -S XALL 1 2 /tmp/foo
    $ ./sar -F -f /tmp/foo
    Linux 3.18.7-100.fc20.x86_64 (localhost.localdomain)    03/13/2015      _x86_64_        (2 CPU)
    
    12:33:27 PM  MBfsfree  MBfsused   %fsused  %ufsused     Ifree     Iused    %Iused FILESYSTEM
    12:33:28 PM      6859     10857     61.28     66.49    901972    259020     22.31 /dev/sda5
    12:33:28 PM       170       113     39.88     46.61     76592       320      0.42 /dev/sda2
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv1
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv2
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv3
    Summary:         6859     10857     61.28     66.49    901972    259020     22.31 /dev/sda5
    Summary:          170       113     39.88     46.61     76592       320      0.42 /dev/sda2
    Summary:           19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv1
    Summary:           19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv2
    Summary:           19         0      0.87      6.03      5125        11      0.21 /dev/mapper/foo-lv3
    $

New option "sar -F MOUNT" gives output as below

    $ ./sar -F MOUNT -f /tmp/foo
    Linux 3.18.7-100.fc20.x86_64 (localhost.localdomain)    03/13/2015      _x86_64_        (2 CPU)
    
    12:33:27 PM  MBfsfree  MBfsused   %fsused  %ufsused     Ifree     Iused    %Iused MOUNTPOINT
    12:33:28 PM      6859     10857     61.28     66.49    901972    259020     22.31 /
    12:33:28 PM       170       113     39.88     46.61     76592       320      0.42 /boot
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /oracle
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /sybase
    12:33:28 PM        19         0      0.87      6.03      5125        11      0.21 /mysql
    Summary:         6859     10857     61.28     66.49    901972    259020     22.31 /
    Summary:          170       113     39.88     46.61     76592       320      0.42 /boot
    Summary:           19         0      0.87      6.03      5125        11      0.21 /oracle
    Summary:           19         0      0.87      6.03      5125        11      0.21 /sybase
    Summary:           19         0      0.87      6.03      5125        11      0.21 /mysql
    $

### CODE REVIEW

**Comment by sysstat** [comment]
Steve,
This new feature is actually interesting.
But before going further, please note the following points (yes I know, I'm a bit fussy :grimacing: ):
* Try to submit several small patches instead of a big one. At least, you can make separate patches for documentation (eg. manual page, or the fact you have chosen -F MOUNT to activate your feature). This is easier for me to review, and maybe I can tell you to update one of your patch if it doesn't please me.
* Also if you have to rework one of your patch after taking into account some suggestions, try to send a pull request which doesn't include all the intermediate commits which have been modified by subsequent ones.
* Last there is a difficult point with sysstat which concerns system activity datafiles format. The idea is to make it possible for a new version of sysstat (adding new kind of statistics data) to read system activity datafiles from previous sysstat versions. Moreover I also try to make it possible for old sysstat versions to be able to read datafiles from newer versions (a sysstat version won't display statistics for new activities added in a future version, but it should be able to read and display statistics for known activities). All this to say that it is always a bit messy to change the system activity datafile format as you have done here.

Thanks a lot anyway for all your contributions. I will look at your patch with more attention later (sorry I can't do this now as I don't have much time) and will get back in touch with you.
Sebastien.

**Comment by jaunix** [comment]
Thanks for the advice.  New around here so the guidance on most suitable approach is very much appreciated. 

The change in file format sounds like a bit of a show-stopper really, perhaps this enhancement needs reconsidering at the time of the next major sysstat release.

**Comment by sysstat** [comment]
In fact some enhancements have been added to sysstat to make it easier to add new statistics or activities while still keeping a compatible format for sa datafiles.
So maybe in your case (adding mount point and/or increasing filesystem name length), this won't be a problem but I need to take some time to check to make sure it's OK. I'll do it ASAP.

File: man/sar.in [hunk 1]
```diff
@@ -348,16 +348,21 @@ Set the ending time of the report. The default ending time is
 18:00:00. Hours must be given in 24-hour format.
 This option can be used when data are read from
 or written to a file (options -f or -o).
-.IP -F
+.IP "-F [ MOUNT ]"
```
Comment (sysstat) [inline man/sar.in]: Parameter MOUNT should also appear in the SYNOPSIS section at the beginning of the man page.

File: pr_stats.c [hunk 2]
```diff
@@ -2500,11 +2500,11 @@ __print_funct_t stub_print_filesystem_stats(struct activity *a, int curr, int di
...
 	struct stats_filesystem *sfc, *sfm;
 
-
 	if (dis) {
 		printf("\n%-11s  MBfsfree  MBfsused   %%fsused  %%ufsused"
-		       "     Ifree     Iused    %%Iused FILESYSTEM\n",
-		       (dispavg ? _("Summary:") : timestamp[!curr]));
+		       "     Ifree     Iused    %%Iused %s\n",
+		       (dispavg ? _("Summary:") : timestamp[!curr]),
+		       DISPLAY_MOUNT(a->opt_flags) ? _("MOUNTPOINT") : _("FILESYSTEM"));
```
Comment (sysstat) [inline pr_stats.c]: MOUNTPOINT and FILESYSTEM should not be enclosed in _() since they are not translated.

File: pr_stats.c [hunk 3]
```diff
@@ -2528,7 +2528,7 @@ __print_funct_t stub_print_filesystem_stats(struct activity *a, int curr, int di
 		       sfc->f_files - sfc->f_ffree,
 		       sfc->f_files ? SP_VALUE(sfc->f_ffree, sfc->f_files, sfc->f_files)
 				    : 0.0,
-		       sfc->fs_name);
+		       DISPLAY_MOUNT(a->opt_flags) ? sfc->mountp : sfc->fs_name);
```
Comment (sysstat) [inline pr_stats.c]: pr_stats.c has been modified to display mountpoints. But there are other files to update for the various output formats used by sadf (json_stats.c, xml_stats.c, rndr_stats.c)

File: sar.c [hunk 4]
```diff
@@ -170,7 +171,7 @@ void display_help(char *progname)
 	printf(_("\t-q\tQueue length and load average statistics\n"));
 	printf(_("\t-R\tMemory statistics\n"));
 	printf(_("\t-r [ ALL ]\n"
-		 "\tMemory utilization statistics\n"));
+		 "\t\tMemory utilization statistics\n"));
```
Comment (sysstat) [inline sar.c]: Well observed! But this modification should not be part of this patch as it is not related to the addition of the "mountpoint" feature.

**Comment by sysstat** [comment]
Steve,
I made some comments on your patch so that you can see what I was expecting.
I plan to pull your request in the coming days anyway (and even make the expected changes :-) )
BTW there are no compatibility problems concerning the format of sar datafiles with the addition of the mountpoint names.

**Comment by jaunix** [comment]
OK, thanks for the feedback.  If you want any of these points addressing via a new pull request, let me know and I'll get them done.

**Comment by sysstat** [comment]
Yes, please send me a new pull request if you have time to do so. Else I plan to work on your patch and make the changes next Friday.
Thanks for your help.

File: rd_stats.h [hunk 5]
```diff
@@ -543,6 +543,7 @@ struct stats_filesystem {
 	unsigned long long f_files		__attribute__ ((aligned (16)));
 	unsigned long long f_ffree		__attribute__ ((aligned (16)));
 	char 		   fs_name[MAX_FS_LEN]	__attribute__ ((aligned (16)));
+	char 		   mountp[MAX_FS_LEN]	__attribute__ ((aligned (16)));
```
Comment (sysstat) [inline rd_stats.h]: Attribute _((aligned (16))) not needed for field "mountp". It will be packed at the end of previous one ("fs_name").

File: pr_stats.c (hunk 3 above)
Comment (jaunix) [inline pr_stats.c]: Thanks, have sent pull requests for these now.  Broke them down as below.
json https://github.com/sysstat/sysstat/pull/53
xml https://github.com/sysstat/sysstat/pull/54
rndr https://github.com/sysstat/sysstat/pull/55
